run_linux.go

```diff
@@ -2419,7 +2440,126 @@ func getSecretMount(tokens []string, secrets map[string]string, mountlabel strin
 	return &newMount, nil
 }
 
-func cleanupRunMounts(paths []string, mountpoint string) error {
+// getSSHMount parses the --mount type=ssh flag in the dockerfile, checks if there's a ssh source provided, and creates and starts an ssh agent to be forwarded into the container
+func getSSHMount(tokens []string, count int, sshsources map[string]*sshagent.Source, mountlabel string, uidmap []spec.LinuxIDMapping, gidmap []spec.LinuxIDMapping, processLabel string) (*spec.Mount, *sshagent.AgentServer, error) {
+	errInvalidSyntax := errors.New("ssh should have syntax id=id[,target=path,required=bool,mode=uint,uid=uint,gid=uint")
+
+	var err error
+	var id, target string
+	var required bool
+	var uid, gid uint32
+	var mode uint32 = 400
+	for _, val := range tokens {
+		kv := strings.SplitN(val, "=", 2)
```

Comment:
If you don't have an equal sign in the token, you'll get some unhappiness.  Might want to do a len check on kv to be completely safe.